src/main/kotlin/no/nb/mlt/wls/infrastructure/callbacks/InventoryNotifierAdapter.kt

```diff
@@ -25,15 +39,35 @@ class InventoryNotifierAdapter(
     }
 
     override fun orderChanged(order: Order) {
+        val payload = objectMapper.writeValueAsString(order.toNotificationOrderPayload())
+        val timestamp = System.currentTimeMillis().toString()
+        val signature = generateSignature(payload, timestamp)
+
         // TODO: Should probably have a more robust retry mechanism, what if receiver is down for a while?
         webClient
             .post()
             .uri(order.callbackUrl)
-            .bodyValue(order.toNotificationOrderPayload())
+            .bodyValue(payload)
+            .header("X-Signature", signature)
+            .header("X-Timestamp", timestamp)
```

Comment:
See similar comment above

```suggestion
            .header("X-Timestamp", timestamp)
            .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
```